src/test_helpers/tests/gold_stream_evaluator_test.cc

```diff
@@ -0,0 +1,168 @@
+}
+
+TEST_F(GoldStreamEvaluatorTest, BadActualStream)
+{
+  gold_iss->setstate(std::ios_base::goodbit);
+  actual_iss->setstate(std::ios_base::badbit);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_FALSE(test_eval.ActualGood());
+  ASSERT_TRUE(test_eval.GoldGood());
+}
+
+TEST_F(GoldStreamEvaluatorTest, BadBothStream)
+{
+  gold_iss->setstate(std::ios_base::badbit);
+  actual_iss->setstate(std::ios_base::badbit);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_FALSE(test_eval.ActualGood());
+  ASSERT_FALSE(test_eval.GoldGood());
+}
+
+TEST_F(GoldStreamEvaluatorTest, SameStream)
+{
+  std::string input_text = "1\n2\n3\n4\n5";
+  gold_iss->str(input_text);
+  actual_iss->str(input_text);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_TRUE(test_eval.Compare());
+}
+
+TEST_F(GoldStreamEvaluatorTest, DiffStream)
+{
+  std::string gold_text = "1\n2\n3\n4\n5";
+  std::string actual_text = "1\n2\nX\n4\n5";
+  gold_iss->str(gold_text);
+  actual_iss->str(actual_text);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_FALSE(test_eval.Compare());
+}
+
+TEST_F(GoldStreamEvaluatorTest, BadGoldStreamCompare)
+{
+  actual_iss->setstate(std::ios_base::goodbit);
+  gold_iss->setstate(std::ios_base::badbit);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_THROW(test_eval.Compare(), std::runtime_error);
+}
+
+TEST_F(GoldStreamEvaluatorTest, BadActualStreamCompare)
+{
+  gold_iss->setstate(std::ios_base::goodbit);
+  actual_iss->setstate(std::ios_base::badbit);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_THROW(test_eval.Compare(), std::runtime_error);
+}
+
+TEST_F(GoldStreamEvaluatorTest, CompareLongerGold)
+{
+  std::string gold_text = "1\n2\n3\n4\n5";
+  std::string actual_text = "1\n2\n3\n4";
+  gold_iss->str(gold_text);
+  actual_iss->str(actual_text);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_FALSE(test_eval.Compare());
+
+}
+
+TEST_F(GoldStreamEvaluatorTest, CompareLongerActual)
+{
+  std::string gold_text = "1\n2\n3\n4";
+  std::string actual_text = "1\n2\n3\n4\n5";
+  gold_iss->str(gold_text);
+  actual_iss->str(actual_text);
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  ASSERT_FALSE(test_eval.Compare());
+}
+
+TEST_F(GoldStreamEvaluatorTest, DiffWorks)
+{
+  std::string gold_text = "1\n2\nX\n4";
+  std::string actual_text = "1\n2\n3\n4";
+  std::string diff;
+  gold_iss->str(gold_text);
+  actual_iss->str(actual_text);
+
+  btest::GoldStreamEvaluator test_eval(std::move(gold_iss), std::move(actual_iss));
+  diff = test_eval.GetDiff();
+  ASSERT_EQ(diff,"@@ -1,4 +1,4 @@\n 1\n 2\n-X\n+3\n 4\n");
+}
+
+TEST_F(GoldStreamEvaluatorTest, DiffWorksSame)
+{
```

Comment:
Same thing about open curly brace.

Reply:
Also, for most cases, function name should have one space before the open parenthesis. The same thing applies to for, while, if etc.